lib/active_merchant/billing/gateways/spreedly_core.rb

```diff
@@ -44,7 +44,7 @@ def purchase(money, payment_method, options = {})
           purchase_with_token(money, payment_method, options)
         else
           MultiResponse.run do |r|
-            r.process { save_card(false, payment_method, options) }
+            r.process { save_card(options[:store], payment_method, options) }
```

Comment:
I like the direction you've gone in with `save_card`. However, I do think that there's an additional refactoring opportunity here. Consider removing `options[:store]` as an argument to `save_card` all together and leverage the existing `options` hash instead. The `save_card` method could then be modified to only accept two arguments and determine when the retained attributed should be true:
```
def save_card(credit_card, options)
  ......existing code.....
  doc.retained(true) if options[:retain]
end
```

Additionally, instead of using `store` as key, `retain` is more appropriate when working our API. Using `store` may lead to confusion with the `store` method itself.

Thoughts?

Reply:
I've updated to maintain `:store` too.

Please advice @dtykocki if it's better to revert back.

Reply:
@abarrak - Let's just go with using `store` instead of supporting both `store` and `retain`. I think it'll keep things clearer this way. Once you've made the change, we should be ready for a merge.

Reply:
Alright, I restored back old setup in new PR #2798.